### ISSUE TITLE
prefer-stateless-function with decorators 

### ISSUE THREAD

**Issue by captify-abogatyrov** [yannickcr/eslint-plugin-react#1034]
i am using decorators, decorators work just with class so i must not use stateless function
```javascript
@dragDropContext(HTML5Backend)
class ToDoList extends Component {
  render() {
    const { data } = this.props;
    return (
      <div>
        {data.map(((todo, i) => <SingleToDo
          data={todo}
        />))}
      </div>
    );
  }
}
```
and in this case i have error prefer-stateless-function

**Comment by ljharb** [comment]
It makes sense that a decorated `class` should not be reported here.

**Comment by OscarBarrett** [comment]
Could a new release be published that includes this please?
